pkg/clusteragent/admission/mutate/agent_sidecar/providers.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package agentsidecar
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	corev1 "k8s.io/api/core/v1"
+)
+
+////////////////////////////////
+//                            //
+//     Provider Overrides     //
+//                            //
+////////////////////////////////
+
+// ProviderIsSupported indicates whether the provider is supported by agent sidecar injection
+func ProviderIsSupported(provider string) bool {
+	switch provider {
+	case providerFargate:
+		return true
+	case "":
+		// case of empty provider
+		return true
+	default:
+		return false
+	}
+}
+
+func applyProviderOverrides(container *corev1.Container) error {
+	provider := config.Datadog.GetString("admission_controller.agent_sidecar.provider")
+
+	if !ProviderIsSupported(provider) {
+		return fmt.Errorf("unsupported provider: %v", provider)
+	}
+
+	switch provider {
+	case providerFargate:
+		return applyFargateOverrides(container)
+	}
+
+	return nil
+}
+
+func applyFargateOverrides(container *corev1.Container) error {
```

Comment:
I'm not sure about this. Would it make sense to have somewhere a map of providers to profileOverrides and a generic function to apply a profile instead of having a specific one per provider?

Reply:
I don't think so.

A provider corresponds to a cluster environment (e.g. fargate, gke autopilot, etc.)
For each provider, we have a specific and predefined set of overrides.

Profiles are, on the other hand, user-provided overrides to customise any default settings.

In the future, we might want to have different flavours for a provider (something like `xs`, `s`, `L`, corresponding to limits/requests for example). Each flavour will have a different set of overrides. At that point, yes it would make sense to have a map.

However, for now, it is not yet clear how this configuration will be defined. For example, are we going to include the flavour in the provider name (like `fargate-xss`, `fargate-xs`, etc.)? Or are we going to have a separate config param to set the flavour (something like `admission_controller.agent_sidecar.provider_flavour: xss` )?

Since this is not yet clear, I think it is better to keep things simplified now to reduce the need for a lot of code refactoring in the future.